packages/compass-connections/src/components/connections.tsx

```diff
@@ -99,12 +97,14 @@ function Connections({
     dataService: DataService
   ) => void;
   isConnected: boolean;
-  connectionStorage?: typeof ConnectionStorage;
   appName: string;
   getAutoConnectInfo?: () => Promise<ConnectionInfo | undefined>;
   connectFn?: ConnectFn;
 }): React.ReactElement {
   const { log, mongoLogId } = useLoggerAndTelemetry('COMPASS-CONNECTIONS');
+  // @TODO: Extract to a prop COMPASS-7397
+  const connectionStorage = connectionStorageLocator();
```

Comment:
Just in case want to clarify that COMPASS-7397 is not about passing services through props, we want to convert compass-connections to a plugin that follows the patterns that other plugins follow: service injection and service access should happen in a dedicated store outside of React lifecycle